File: tests/compatibility/v0.5.3/swift/compatibility.swift
import Foundation
import Uimd

public final class CompatibilityApp: CompatibilityUI
{
    public override init()
    {
        super.init()
        scroller.setChild(CompatScrollUI())
    }
}

@main
enum Program
{
    static func main()
    {
        let code = GeneratedWindowRuntime.runGeneratedAppMain
        {
            let app = CompatibilityApp()
            return GeneratedWindowRuntime.runGeneratedWindow(app, app.runtimeOptions(), CommandLine.arguments)
        }
        Foundation.exit(Int32(code))
    }
}

File: tests/compatibility/v0.5.3/go/compatibility.go
package main

import (
    "os"

    uimd "uimd"
)

type CompatibilityApp struct {
    *CompatibilityUI
}

func NewCompatibilityApp() *CompatibilityApp {
    app := &CompatibilityApp{
        CompatibilityUI: NewCompatibilityUI(),
    }
    app.SetEventHandler(app)
    app.Scroller.SetChild(NewCompatScrollUI())
    return app
}

func main() {
    code := uimd.RunGeneratedAppMain(func() int {
        app := NewCompatibilityApp()
        return uimd.RunGeneratedWindow(app, app.RuntimeOptions(), os.Args)
    })
    os.Exit(code)
}
